### PR TITLE
refactor(editor): Port more components over to composition API (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
This PR migrates some more design-system components over to the new component style, and also brings down the number of type-errors in design-system from **309** down to **40**.

## Review / Merge checklist
- [x] PR title and summary are descriptive

### CODE REVIEW

File: packages/design-system/src/components/N8nDatatable/Datatable.vue [hunk 1]
```diff
@@ -175,6 +60,89 @@ export default defineComponent({
...
+	if (maxPage < props.currentPage) {
+		onUpdateCurrentPage(maxPage);
+	}
+}
+
+function getTdValue(row: DatatableRow, column: DatatableColumn) {
+	return getValueByPath<DatatableRowDataType>(row, column.path);
+}
+
+function getThStyle(column: DatatableColumn) {
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nDatatable/Datatable.vue:139]: nitpick - would be nice to fix the typo

File: packages/design-system/src/components/N8nNodeIcon/NodeIcon.vue [hunk 2]
```diff
@@ -39,6 +39,7 @@
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
 import { defineComponent, type PropType } from 'vue';
 import N8nTooltip from '../N8nTooltip';
+import { Placement } from 'element-plus';
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nNodeIcon/NodeIcon.vue:42]:
```suggestion
import type { Placement } from 'element-plus';
```

**Review by mutdmour — DISMISSED** [review]
LGTM. Much cleaner now. Just a couple of minor comments.

File: packages/design-system/src/components/N8nDatatable/Datatable.vue (hunk 1 above)
Comment (netroy) [reply packages/design-system/src/components/N8nDatatable/Datatable.vue:139]: `getThStyle` returns the style for a `th` element. So, this is not a misspelt `getTheStyle`.

Comment (mutdmour) [reply packages/design-system/src/components/N8nDatatable/Datatable.vue:139]: ah misunderstood it 😅

**Review by mutdmour — DISMISSED** [review]

**Comment by netroy** [comment]
Interesting that the E2E tests were [passing](https://github.com/n8n-io/n8n/actions/runs/8124718420) before I opened the PR, but now they fail 😕.

**Review by alexgrozav — APPROVED** [review]
Amazing work, @netroy!

**Comment by janober** [comment]
Got released with [`n8n@1.34.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.34.0)

**Pull request merged**